### ISSUE TITLE
FINOS to Provide GCP Access to Compliant Financial Infrastructure

### ISSUE THREAD

**Issue by mcleo-d** [finos/compliant-financial-infrastructure#219]
@AdrianHammond and @peter-thomas-db,

Can you help lead the conversation on this issue with @maoo on what is needed for #201.

Many thanks,

James.

**Comment by maoo** [comment]
Thanks @mcleo-d for involving me!

We can create an account on https://console.cloud.google.com and share credentials with the team member that will take care of configuring the services needed (we've already done this for other FINOS projects). In order to proceed, I'd like to know:
- name, email address and preferably slack handle of the team member that will take care of this setup; this way we can establish a direct collaboration channel and work efficiently
- List of GCP services you'd like to use
- A monthly cost estimate, as I'll need to request budget for it; given the nature of the integration, I do not expect high costs, and therefore I don't foresee issues with budget either

Feel free to submit this info to help@finos.org , as there may be sensitive data (ie email addresses), and we'll get back to you. Thank you!

**Comment by AdrianHammond** [comment]
Hi @maoo 

Re your questions:

- name, email address and preferably slack handle of the team member that will take care of this setup; this way we can establish a direct collaboration channel and work efficiently

I am happy to assist with this for any OpenShift installations on GCP, I don't have much experience of GCP outside of this. I have sent my email to help@finos.org

- List of GCP services you'd like to use

OpenShift install only requires basic IaaS services e.g. compute, storage etc. Full list is [here](https://docs.openshift.com/container-platform/4.10/installing/installing_gcp/installing-gcp-account.html#installation-gcp-limits_installing-gcp-account) in the OCP doc.

- A monthly cost estimate, as I'll need to request budget for it; given the nature of the integration, I do not expect high costs, and therefore I don't foresee issues with budget either

I am not certain how and who would be using the account so it's difficult for me to answer. For the testing I am doing for CFI I am using a minimal OCP install with six nodes (n1-standard-4), 3 for the control plane and 3 compute nodes. I am using very little storage as I am not using it to run any workload just testing for CFI project. Checking my GCP bill its £150 so far this month. I am stopping the nodes when I am not using the cluster so that is reducing the cost. 

Does that help, happy to answer more questions if needed?

-A-

**Comment by maoo** [comment]
Thanks @AdrianHammond , yes that helps.

If I understand correctly, following the same work done by CFI on EKS, this cluster should bootstrap and tear down within the context of the GitHub Action, which means that we'd be charged only for the time that the test would need to run; is that correct?

In order to create a GCP account, we'd need to create a Gmail account (ie `finos.cfi.dev@gmail.com`) and use that to create a GCP account; at that point, it should be just a matter of setting up some API tokens (And related accounts) with the right permissions, and set them up as secrets into GitHub.

Please let me know if it all makes sense or if I missed something.

Thank you!

**Comment by AdrianHammond** [comment]
Hi @maoo 

I am not familiar with what has been done with EKS. The current status of the OpenShift work on GCP for CFI is a manual process so initially cluster teardown / hibernation would be manual rather than automated. 

-A-

**Comment by maoo** [comment]
Thanks @AdrianHammond . @abdullahgarcia @eddie-knight - could you please my comment above and confirm if my understanding is correct? TY! /cc @mcleo-d

**Comment by mcleo-d** [comment]
Hi @maoo,

Regarding your question https://github.com/finos/compliant-financial-infrastructure/issues/219#issuecomment-1133969486

Development of this process will likely be a 2 phase approach as @AdrianHammond and the team are manually documenting and testing the process before automation tests are written. 

`Phase 1` means a GCP environment for manual creation and destruction will initially be needed. The instance won't be kept running after test though. @AdrianHammond and @peter-thomas-db can help confirm this.

`Phase 2` is then automating the `create`, `test` and `destroy` within the context of a GitHub Action. Where again, there won't be a permanent instance running.

I hope this helps?

James.

**Comment by AdrianHammond** [comment]
Re "Phase 1 means a GCP environment for manual creation and destruction will initially be needed. The instance won't be kept running after test though. @AdrianHammond and @peter-thomas-db can help confirm this."

We will need to document a procedure to either hibernate or destroy the cluster. I am happy to help with this. 

-A-

**Comment by maoo** [comment]
Sorry for the late reply here! I've created a Google account called finos.cfi@gmail.com , which you'll be able to use to access Google Cloud Platform on cloud.google.com . @mcleo-d , the password can be found in our password vault (search for `FINOS CFI Google Account`.

In order to complete the setup, we need to generate a credit card on Spenddesk, then access cloud.google.com to specify it there. @TheJuanAndOnly99 - can you please take care of it and notify @mcleo-d when it's all done?

Thanks!

**Comment by TheJuanAndOnly99** [comment]
@maoo @mcleo-d waiting on approval for the spendesk card, will add it as soon as it gets approved! Will keep you posted.

**Comment by TheJuanAndOnly99** [comment]
@maoo @mcleo-d the card has been generated and added to the finos.cfi@gmail.com Google Cloud Platform account.

**Comment by maoo** [comment]
Thanks @TheJuanAndOnly99 !

@AdrianHammond @abdullahgarcia @eddie-knight @thinkl33t @peter-thomas-db - we're ready to grant you access to the account, feel free to ping me privately on Slack, so we can coordinate on how to setup the account password as a GitHub Secret; that will allow you to run commands via GitHub Actions.

Right now the account offers some free credits, which I hope will suffice to cover the setup part; when it's all set, we can calculate the expected monthly cost and update our cost reports.

Please let me know if you have any questions.

**Comment by mcleo-d** [comment]
@AdrianHammond and @peter-thomas-db 👋🏻 

Have you moved this forward with @maoo and @TheJuanAndOnly99 as it would be awesome to close this issue?

Kind regards,

James.

cc @abdullahgarcia @eddie-knight @thinkl33t

**Comment by AdrianHammond** [comment]
> @AdrianHammond and @peter-thomas-db 👋🏻
> 
> Have you moved this forward with @maoo and @TheJuanAndOnly99 as it would be awesome to close this issue?
> 
> Kind regards,
> 
> James.
> 
> cc @abdullahgarcia @eddie-knight @thinkl33t

Hi @mcleo-d  - on vacation last week, I will do my best to pick this up this week.

**Comment by maoo** [comment]
Hi @AdrianHammond ! I'm happy to contact the Sales team, but I'd need to know how much you'd need quotas to increase. Could you please provide me with some numbers? TY!

**Comment by AdrianHammond** [comment]
> Hi @AdrianHammond ! I'm happy to contact the Sales team, but I'd need to know how much you'd need quotas to increase. Could you please provide me with some numbers? TY!

Hi  - 72 CPU's and 1Tb disk please.

**Comment by maoo** [comment]
Wow, that seems like a lot of CPUs! Is there any chance we can start with a smaller setup?

**Comment by maoo** [comment]
I'd turn the question around: what's the bare minimum? 4? 8?

**Comment by AdrianHammond** [comment]
3 x control plane nodes and 3 x compute nodes each with 4 CPU's = 24.

**Comment by AdrianHammond** [comment]
@maoo had quota increased to 12 vCPUs. Tested that I could spin cluster up with the new quota.
